apps/emqx_machine/src/emqx_machine.erl

```diff
@@ -0,0 +1,181 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_machine).
+
+-export([ start/0
+        , graceful_shutdown/0
+        , is_ready/0
+        ]).
+
+-export([ stop_apps/1
+        , ensure_apps_started/0
+        ]).
+
+-export([sorted_reboot_apps/0]).
+
+-include_lib("emqx/include/logger.hrl").
+
+%% @doc EMQ X boot entrypoint.
+start() ->
+    os:set_signal(sighup, ignore),
+    os:set_signal(sigterm, handle), %% default is handle
+    ok = set_backtrace_depth(),
+    ok = print_otp_version_warning(),
+
+    %% need to load some app envs
+    %% TODO delete it once emqx boot does not depend on modules envs
+    _ = load_modules(),
+    ok = load_config_files(),
+
+    ok = ensure_apps_started(),
+
+    _ = emqx_plugins:load(),
+
+    ok = print_vsn(),
+
+    ok = start_autocluster(),
+    %% NOTE: keep this to the end
```

Comment:
In my opinion， 
1. If we want to keep this to the end, we should move this call out of `emqx_machine:start/0`.
```erlang
emqx_machine:start(),
emqx_machine_terminator:start().
```
2. I find emqx_machine_terminator link to application control, not under emqx_machine_sup, any reason for this?
I think emqx_machine_sup should monitor it by one_for_one, restart  => transient.

Reply:
1. agreed, will change.
2. one concerns: if `emqx_machine_terminator` is under supervision tree, the call to `init:stop()` will kill itself as a part of `emqx_machine` app, but not *after* `emqx_machine`.

Reply:
a second though: the exist of `emqx_machine` app is probably enough to indicate the node being down since the chance for otp apps to hang is very small.
changed it to reside under supervisor.